src/backend/commands/variable.c

```diff
@@ -933,3 +935,97 @@ show_role(void)
 	/* Otherwise we can just use the GUC string */
 	return role_string ? role_string : "none";
 }
+
+
+/* check_hook: validate new log_connection_messages value */
+bool
+check_log_connection_messages(char **newval, void **extra, GucSource source)
+{
+	char		*rawname;
+	List		*namelist;
+	ListCell	*l;
+	char		*log_connection_messages = *newval;
+	bool		*myextra;
+
+	/*
+	 * Set up the "extra" struct actually used by assign_log_connection_messages.
+	 */
+	myextra = (bool *) malloc(4 * sizeof(bool));
+	if (!myextra)
+		return false;
+	memset(myextra, 0, 4 * sizeof(bool));
+
+	*extra = (void *) myextra;
+
+	if (log_connection_messages == NULL || log_connection_messages[0] == '\0') {
+		return true;
+	}
+
+	if (strcmp(*newval, "all") == 0)
```

Comment:
```suggestion
	if (pg_strcasecmp(*newval, "all") == 0)
```
Case-independent is better here and further, I guess

Reply:
I see both `pg_strcasecmp` and `strcmp` are used [src/backend/commands/variable.c](https://github.com/sdudoladov/postgres/pull/7/files/471f295df8a473d92e9389357f37d6d166866790#diff-9bee4855e89439f420d8845f9c5ff4a9df8da1e186c0a6b1e3de24d2f17869a4).

In that particular case `SplitIdentifierString` actually does the downcasing but I've swtiched to `pg_strcasecmp` anyway